### ISSUE TITLE
NullPointerException when submitting transaction to create account.

### ISSUE THREAD

**Issue by chainhead** [stellar/java-stellar-sdk#38]
I am using the following to create a new account.

```
	public void submitCreateAccountTxn(String fromAccId, String toAccId, String amt, String memo)
			throws IOException {
		Network.use(new Network(NETWORK_PASSPHRASE));
		Server server = new Server(HORIZON_END_POINT);
		KeyPair source = KeyPair.fromAccountId(fromAccId);
		KeyPair destination = KeyPair.fromAccountId(toAccId);
		
		AccountResponse sourceAccount = server.accounts().account(source);
		Transaction txn = new Transaction.Builder(sourceAccount)
				.addOperation(new CreateAccountOperation.Builder(destination, amt).build()).addMemo(Memo.text(memo))
				.build();
		txn.sign(source);

		SubmitTransactionResponse response = server.submitTransaction(txn);
}
```

It fails as follows:

```
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.jdt.internal.jarinjarloader.JarRsrcLoader.main(JarRsrcLoader.java:58)
Caused by: java.lang.NullPointerException
	at org.stellar.sdk.xdr.Signature.encode(Signature.java:23)
	at org.stellar.sdk.xdr.DecoratedSignature.encode(DecoratedSignature.java:36)
	at org.stellar.sdk.xdr.TransactionEnvelope.encode(TransactionEnvelope.java:42)
	at org.stellar.sdk.Transaction.toEnvelopeXdrBase64(Transaction.java:191)
	at org.stellar.sdk.Server.submitTransaction(Server.java:120)
	... 8 more

```

What am I missing?

**Comment by bartekn** [comment]
You are creating `Keypair` without a secret key:
```java
KeyPair source = KeyPair.fromAccountId(fromAccId);
```
So it's unable to sign a transaction. Use [Keypair.fromSecretSeed](https://stellar.github.io/java-stellar-sdk/org/stellar/sdk/KeyPair.html#fromSecretSeed-char:A-).

I'm leaving this issue open to improve error message.

**Comment by chainhead** [comment]
Ok. So, what does the `KeyPair.fromAccountId()` return then?

**Comment by bartekn** [comment]
A KeyPair with a public key only. You can use [Keypair.canSign](https://stellar.github.io/java-stellar-sdk/org/stellar/sdk/KeyPair.html#canSign--) to check if the given key pair can sign transactions.

**Comment by chainhead** [comment]
> A `KeyPair` with a public key **only**. 

And, yet the class is named as a `type of pair - misleading! What is even more confusing is, for the same return type, `KeyPair`, `fromAccountId()` and `fromSecretSeed()` return different things.

Is it not possible to do a transaction with public keys only such that, relevant methods figure out the secret code for signing?

**Comment by bartekn** [comment]
Fixed in [0.1.13](https://github.com/stellar/java-stellar-sdk/releases/tag/0.1.13). Now `KeyPair.sign` throws an exception with more useful information.
